frontend/src/pages/CompareV2.tsx

```diff
@@ -220,9 +233,17 @@ function VisualizationPlaceholder(props: VisualizationPlaceholderProps) {
   );
 }
 
+interface SelectedArtifact {
+  selectedItem: SelectedItem;
+  selectedArtifact?: Artifact;
+}
+
 interface MetricsDropdownProps {
   filteredRunArtifacts: RunArtifact[];
+  metricsTab: MetricsType;
   metricsTabText: string;
+  selectedArtifacts: { [key: string]: SelectedArtifact[] };
+  setSelectedArtifacts: (selectedArtifacts: { [key: string]: SelectedArtifact[] }) => void;
```

Comment:
We are passing more data than individual MetricsDropdown needs here. Each MetricsDropdown only need one `SelectedArtifact[]` but we are passing a map to props. It can make props overly complicated and there is a risk of data mutation in another metricsTab. How about the following logic:

```
interface MetricsDropdownProps {
  filteredRunArtifacts: RunArtifact[];
  metricsTabText: string;
  selectedArtifacts: SelectedArtifact[];
  updateSelectedArtifacts: (selectedArtifacts: SelectedArtifact[]) => void;
```

See my other comment on the caller's side.

Reply:
This sounds good! I have modified the `MetricsDropdownProps` to match this suggestion, except I still need to include the `metricsTab` parameter to know which visualization I am supposed to show. Right now, this is whether or not I should display the confusion matrix, but in the subsequent PR, this will be whether to display the confusion matrix, HTML, or Markdown.

I could simplify this by removing the `metricsTabText` prop and simply re-calculating that within this component, it depends on whether that seems like a simplification or another unnecessary calculation. Do you have thoughts on that?

Reply:
I see, feel free to keep `metricsTab` as it is a valid use of props. It is okay not to remove `metricsTabText`.